### PR TITLE
Update webserver.rst to include information of supported OAuth2 providers

### LINKED ISSUE #35131: Support for general OIDC providers or making it clear in document
### Description

I tried hard to configure airflow with authentik OIDC but airflow kept complaining about empty userinfo. There are very limited tutorials online. After reading some source code of authlib, flask-appbuilder and airflow, I found in [airflow/airflow/auth/managers/fab/security_manager/override.py](https://github.com/apache/airflow/blob/ef497bc3412273c3a45f43f40e69c9520c7cc74c/airflow/auth/managers/fab/security_manager/override.py) that only a selection of providers are supported (github twitter linkedin google azure openshift okta keycloak). If the provider name is not within this list, it will always return an empty userinfo at [line 1475](https://github.com/apache/airflow/blob/ef497bc3412273c3a45f43f40e69c9520c7cc74c/airflow/auth/managers/fab/security_manager/override.py#L1475C22-L1475C22). 

For others who try to integrate openid connect, I would recommend read the code in [airflow/airflow/auth/managers/fab/security_manager/override.py starting from line 1398](https://github.com/apache/airflow/blob/ef497bc3412273c3a45f43f40e69c9520c7cc74c/airflow/auth/managers/fab/security_manager/override.py#L1398)

### Use case/motivation

This behaviour should be documented. Otherwise, there should be a way to configure other OIDC providers like other projects that support OIDC in a general manner. 

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by rqi14**
To include information of supported OAuth2 providers
Closes: #35131 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Comment by vincbeck** [comment]
Static checks are failing. You can fix them by running pre-commits. Please see [documentation](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)

**Comment by rqi14** [comment]
> Static checks are failing. You can fix them by running pre-commits. Please see [documentation](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)

It believes `keycloak` has spelling error. Is there a way to fix?

File: docs/apache-airflow/security/webserver.rst [hunk 1]
```diff
@@ -95,7 +95,9 @@ Other Methods
 
 Since Airflow 2.0, the default UI is the Flask App Builder RBAC. A ``webserver_config.py`` configuration file
 is automatically generated and can be used to configure the Airflow to support authentication
-methods like OAuth, OpenID, LDAP, REMOTE_USER.
+methods like OAuth, OpenID, LDAP, REMOTE_USER. It should be noted that due to the limitation of Flask AppBuilder 
+and Authlib, only a selection of OAuth2 providers is supported. This list includes github, githublocal, twitter, 
+linkedin, google, azure, openshift, okta, keycloak and keycloak_before_17.
```
Comment (vincbeck) [inline docs/apache-airflow/security/webserver.rst:100]:
```suggestion
and Authlib, only a selection of OAuth2 providers is supported. This list includes Github, Githublocal, Twitter, 
Linkedin, Google, Azure, Openshift, Okta, Keycloak and Keycloak_before_17.
```

**Comment by rqi14** [comment]
> > > Static checks are failing. You can fix them by running pre-commits. Please see [documentation](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)
> > 
> > 
> > It believes `keycloak` has spelling error. Is there a way to fix?
> 
> Regardless of whether it is going to fix it, we probably should capitalize them

I guess it is not a very good idea to capitalize them. I left them as it is in the code. If the capitalized name is put into the setting as one reads the document, the code will not recognize the provider and return an error.

File: docs/apache-airflow/security/webserver.rst (hunk 1 above)
Comment (vincbeck) [inline docs/apache-airflow/security/webserver.rst]:
```suggestion
and Authlib, only a selection of OAuth2 providers is supported. This list includes ``github``, ``githublocal``, ``twitter``, 
``linkedin``, ``google``, ``azure``, ``openshift``, ``okta``, ``keycloak`` and ``keycloak_before_17``.
```

**Comment by vincbeck** [comment]
> > > > Static checks are failing. You can fix them by running pre-commits. Please see [documentation](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)
> > > 
> > > 
> > > It believes `keycloak` has spelling error. Is there a way to fix?
> > 
> > 
> > Regardless of whether it is going to fix it, we probably should capitalize them
> 
> I guess it is not a very good idea to capitalize them. I left them as it is in the code. If the capitalized name is put into the setting as one reads the document, the code will not recognize the provider and return an error.

Very good point! Then we should wrap these values with "``<value>``". See my suggestion

**Pull request merged**